Subscribers/Lifecycle.php

```diff
@@ -167,6 +169,29 @@ public function onDeleteArticle(\Enlight_Event_EventArgs $eventArgs)
     }
 
     /**
+     * Callback function to delete an product from connect
+     * after it is going to be deleted locally
+     *
+     * @param \Enlight_Event_EventArgs $eventArgs
+     */
+    public function onPreDeleteCategory(\Enlight_Event_EventArgs $eventArgs)
+    {
+        $category = $eventArgs->get('entity');
+        $this->connectExport->markProductsInToBeDeletedCategories($category);
+    }
+
+    /**
+     * Callback function to delete an product from connect
+     * after it is going to be deleted locally
+     *
+     * @param \Enlight_Event_EventArgs $eventArgs
+     */
+    public function onPostDeleteCategory(\Enlight_Event_EventArgs $eventArgs)
+    {
+        $this->connectExport->handleMarkedProducts();
```

Comment:
In preDelete you already stored categoryId. So in this function you need to check "autoUpdateProducts" config.
```$autoUpdateProducts = $this->config->getConfig('autoUpdateProducts', Config::UPDATE_AUTO);``` can find examples in other subscribers. this config has 3 different values. UPDATE_AUTO means every change of product/variant must be handled. UPDATE_CRON_JOB means changed products will be updated with cron_update flag = 1 and Connect cronjob will generate necessary changes later  and UPDATE_MANUAL means Connect plugin doesn't need to take care of changed products. Customer will update them manually. 

**1.** When values of **autoUpdateProducts** is UPDATE_MANUAL you just need to call return;

**2.** In other 2 cases you need  to fetch all child categories as you did it in onPreDeleteCategory. After that find all related articleIds.

**3.** When values of **autoUpdateProducts** is UPDATE_AUTO then generate changes for all related articleIds ```$this->export($articlesToBeExported);```

**4.** When values of **autoUpdateProducts** is UPDATE_CRON_JOB then update cron_update = 1 for related articledIds.

Reply:
the lifecycleProvider gets the autoUpdateProducts value from its constructor
no need to call config but i have to take care of that